polkadot/xcm/src/v4/mod.rs

```diff
@@ -1314,8 +1314,22 @@ impl<Call: Decode + GetDispatchInfo> TryFrom<NewInstruction<Call>> for Instructi
 			HrmpChannelAccepted { recipient } => Self::HrmpChannelAccepted { recipient },
 			HrmpChannelClosing { initiator, sender, recipient } =>
 				Self::HrmpChannelClosing { initiator, sender, recipient },
-			Transact { origin_kind, mut call } => {
-				let require_weight_at_most = call.take_decoded()?.get_dispatch_info().call_weight;
+			Transact { origin_kind, mut call, fallback_max_weight } => {
+				// We first try to decode the call, if we can't, we use the fallback weight,
+				// if there's no fallback, we just return `Weight::MAX`.
+				let require_weight_at_most = match call.take_decoded() {
+					Ok(decoded) => decoded.get_dispatch_info().call_weight,
+					Err(error) => {
+						let fallback_weight = fallback_max_weight.unwrap_or(Weight::MAX);
+						log::error!(
```

Comment:
Should this one be an error or a warning?

Reply:
Since we expect this particular kind of error, I thought it would be reasonable to downgrade it to a warning...

Reply:
Nice catch! Should actually be a debug